docs/triply-api/index.md

```diff
@@ -824,8 +824,233 @@ Result:
 
+```graphql
+type Book {
+  id:ID!
+  title:[XsdString]!
+}
+```
+This object type corresponds to the shape below:
+
+```turtle
+shp:Book a sh:NodeShape;
+         sh:targetClass sdo:Book;
+         sh:property [ 
+           sh:path dc:title;
+           sh:datatype xsd:string.]
+```
+#### Fields
+Fields in object types, such as `title`, represent properties of nodes. By default, fields return arrays of values. The only exception is when the property has `sh:maxCount: 1`, then the field returns a single value. Then the object type will be:
+
+```graphql
+type Book {
+  id:ID!
+  title:XsdString
+}
+```
+Additionally, following the [best practices](https://graphql.org/learn/best-practices/#nullability), fields can give null results, except for:
+
+- IDs, which represents the IRI of the resource.
+- Lists, but not their elements
+- Properties that have sh:minCount 1 and sh:maxCount 1
+
+Thus, for this shape:
+```turtle
+shp:Book a sh:NodeShape;
+         sh:targetClass sdo:Book;
+         sh:property [ 
+           sh:path dc:title;
+           sh:maxCount "1"^^xsd:integer;
+           sh:minCount "1"^^xsd:integer;
+           sh:datatype xsd:string.]
+```
+The corresponding object type is:
+```graphql
+type Book {
+  id:ID!
+  title:XsdString!
+}
+```
+
+
+If the property shape includes an `sh:datatype`, the field returns values of GraphQL scalar type (see example above). On the other hand, if the property shape has an `sh:class` pointing to a class that:
+- is the `sh:targetClass` of a node shape, the field returns values of the corresponding object type.
+- that is not mentioned as a `sh:targetClass` in a node shape, then the type of the returned values is `ExternalIri`.
+Thus, the shapes :
+```turtle
+shp:Book a sh:NodeShape;
+         sh:targetClass sdo:Book;
+         sh:property [ 
+           sh:path sdo:author;
+           sh:class sdo:Person.];
+         sh:property [ 
+           sh:path sdo:audio;
+           sh:class sdo:AudioObject.].
+
+
+shp:Person a sh:NodeShape;
+         sh:targetClass sdo:Person;
+         sh:property [ 
+           sh:path sdo:name;
+           sh:datatype xsd:string.].
+```
+correspond to the below graphql types:
+```graphql
+type Book {
+  id:ID!
+  author:[Person]!
+  audio:[ExternalIri]!
+}
+
+type Person {
+  id:ID!
+  name:[XsdString]!
+}
+```
+#### IDs 
+IDs represent the IRI of each object. This ID is unique.
+#### Naming
+In order to name the GraphQL types in correspondence to shapes, we follow the below conventions:
+- For object types, we use the `sh:targetClass` of the node shape.
+- For object type fields, we use the `sh:path` of the property shape.
+
+More specifically, the name comes from the part of the IRI after the last `#` or otherwise the last `/`, converted from kebab-case to camelCase. 
+
+Notice that if the selected name is illegal or causes a name collision, we'll return an error informing the user about the problem and ignore this type or field. 
```

Comment:
This could change in the future. Please help remember that we should then also update the documentation!